### PR TITLE
sched: Remove the unnecessary weak_function

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
because not all compiler support the weak attribute, and
many features are either always used or guarded by config.

## Impact
no major difference with stm32f103-minimum:hello
before:
```
   text    data     bss     dec     hex filename
  20043     152    1372   21567    543f nuttx
```
after change:
```
   text    data     bss     dec     hex filename
  20011     152    1372   21535    541f nuttx
```

## Testing
Pass CI

### CODE REVIEW

**Review by pkarashchenko — COMMENTED** [review]
in general looks good

File: sched/signal/signal.h [hunk 1]
```diff
@@ -147,7 +147,7 @@ struct task_group_s;
 
 /* sig_initializee.c */
 
-void weak_function nxsig_initialize(void);
+void               nxsig_initialize(void);
```
[the remark this answers is not in the document]
Comment (xiaoxiang781216) [reply sched/signal/signal.h:150]: keep as before to align the code after line 154

**Review by pkarashchenko — APPROVED** [review]

**Comment by pkarashchenko** [comment]
@acassis could you please take a look and merge?

**Review by pkarashchenko — COMMENTED** [review]
@xiaoxiang781216 could you please add `FAR` to `unsigned int *prio` in `nxmq_do_receive` since we are changing this file?

**Comment by xiaoxiang781216** [comment]
> @xiaoxiang781216 could you please add `FAR` to `unsigned int *prio` in `nxmq_do_receive` since we are changing this file?

Done.

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**